### PR TITLE
Adds clip_grad_norm to all recipe config that supports it

### PR DESCRIPTION
**Pull request by thomasjpfan**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [x] other (Updates config)

Fixes https://github.com/pytorch/torchtune/issues/1993

I ran the following script to find the recipes the support `clip_grad_norm` and update the yaml files.

```python
from torchtune._recipe_registry import get_all_recipes
import torchtune
import os
from pathlib import Path

recipes = get_all_recipes()
recipes_dir = Path(torchtune.__path__[0]).parent / "recipes"
config_dir = recipes_dir / "configs"

support_clip_grads = [
    recipe
    for recipe in recipes
    if "clip_grad_norm" in (recipes_dir / recipe.file_path).read_text()
]

for recipe in support_clip_grads:
    for config in recipe.configs:
        if config.file_path.startswith("dev"):
            yaml_config = recipes_dir / config.file_path
        else:
            yaml_config = config_dir / config.file_path

        assert yaml_config.exists(), yaml_config

        yaml_text = yaml_config.read_text()

        if "loss" not in yaml_text:
            continue

        if "clip_grad_norm" in yaml_text:
            continue

        assert "compile" in yaml_text

        text = yaml_text.split(os.linesep)

        compile_idx = None
        for idx, line in enumerate(text):
            if line.startswith("compile"):
                compile_idx = idx
                break
        assert compile_idx is not None

        text.insert(compile_idx, "clip_grad_norm: null")

        result = os.linesep.join(text)
        with yaml_config.open("w") as f:
            f.write(result)
```

#### Changelog
What are the changes made in this PR?
* Adds `clip_grad_norm` to all recipes that support it.

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [x] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Review by joecummings — APPROVED** [review]
Awesome! Thanks for doing this :)

**Comment by ebsmothers** [comment]
Maybe I am missing something but I don't see gradient clipping support in DPO, PPO, or distributed KD recipes. Can we remove the `clip_grad_norm` fields from those configs? (Separately we should probably enable for KD at the very least but can likely do that separately)

**Comment by thomasjpfan** [comment]
@ebsmothers Thanks for catching the issue! There was a bug in my script for finding the recipes the support gradient clipping. I updated this PR to fix the issue.
